app/src/main/java/ch/epfl/sweng/swenggolf/ListOfferTouchListener.java

```diff
@@ -0,0 +1,74 @@
+package ch.epfl.sweng.swenggolf;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+public class ListOfferTouchListener implements RecyclerView.OnItemTouchListener {
+
+    private final OnItemClickListener mListener;
+
+    public interface OnItemClickListener {
+        void onItemClick(View view, int position);
+
+        void onLongItemClick(View view, int position);
+    }
+
+    private final GestureDetector mGestureDetector;
+
+    /**
+     * Constructs a ListOfferTouchListener, used to handle touch actions on the
+     * RecyclerView of the Offers.
+     *
+     * @param context      the context
+     * @param recyclerView the RecyclerView used
+     * @param listener     the click listener
+     */
+    public ListOfferTouchListener(Context context,
+                                  final RecyclerView recyclerView,
+                                  OnItemClickListener listener) {
+        mListener = listener;
+
+        mGestureDetector =
+                new GestureDetector(context,
+                        new GestureDetector.SimpleOnGestureListener() {
+
+                            @Override
+                            public boolean onSingleTapUp(MotionEvent event) {
+                                return true;
+                            }
+
+                            @Override
+                            public void onLongPress(MotionEvent event) {
+                                View child = recyclerView
+                                        .findChildViewUnder(event.getX(), event.getY());
+                                if (child != null && mListener != null) {
+                                    mListener.onLongItemClick(child, recyclerView
+                                            .getChildAdapterPosition(child));
+                                }
+                            }
+                        });
+    }
+
+    @Override
+    public boolean onInterceptTouchEvent(RecyclerView view, MotionEvent event) {
+        View childView = view.findChildViewUnder(event.getX(), event.getY());
+        if (childView != null && mListener != null && mGestureDetector.onTouchEvent(event)) {
+            mListener.onItemClick(childView, view.getChildAdapterPosition(childView));
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void onTouchEvent(RecyclerView view, MotionEvent motionEvent) {
+        // No specific gestures to be handled except for the click
+    }
+
+    @Override
+    public void onRequestDisallowInterceptTouchEvent(boolean disallowIntercept) {
+        // No gestures to be intercepted before click
+    }
+}
```

Comment:
What are the advantages to use an touchListener here ? You could just set an onClickListener on every view when creating the viewHolder. Maybe you want to use advanced touch control later ?

Reply:
I imagine we could use a swipe left/right to save offers, contact the author, etc. directly on the list. What do you think?

Reply:
These are good idea. We could also expand the offer on a long touch to show full description.

Reply:
Of course, for now it just calls the short touch, but that's the idea.